src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -736,24 +741,13 @@ private void InsertOutput(IEnumerable<string> output, int position)
                 ApplyProtection(_window._outputBuffer, _outputProtection);
             }
 
-            private void FinishExecute(Task<ExecutionResult> result)
+            private void FinishExecute(bool succeeded)
             {
-                // The finished task has been replaced by another task (e.g. reset).
-                // Do not perform any task finalization, it will be done by the replacement task.
-                if (_currentTask != result)
-                {
-                    return;
-                }
-
-                _currentTask = null;
                 ResetCursor();
 
-                if (result.Exception != null || !result.Result.IsSuccessful)
```

Comment:
I believe that such exceptions will now be propagated, which seems desirable since they're from our code, rather than user code (as far as I can tell).